vendor/bat-native-ledger/src/bat/ledger/internal/media/twitch.cc

```diff
@@ -400,99 +390,82 @@ void MediaTwitch::OnMediaPublisherInfo(
+  twitch_events[media_key] = new_event;
 
-    auto callback = std::bind(&MediaTwitch::OnSaveMediaVisit,
-                              this,
-                              _1,
-                              _2);
-
-    ledger_->SaveMediaVisit(publisher_key,
-                            updated_visit_data,
-                            real_duration,
-                            window_id,
-                            callback);
-    ledger_->SetMediaPublisherInfo(media_key, publisher_key);
-  } else {
-    ledger::VisitData updated_visit_data(visit_data);
-    updated_visit_data.name = publisher_info->name;
-    updated_visit_data.url = publisher_info->url;
-    updated_visit_data.provider = TWITCH_MEDIA_TYPE;
-    updated_visit_data.favicon_url = publisher_info->favicon_url;
+  if (real_duration == 0) {
+    return;
+  }
 
-    ledger::TwitchEventInfo old_event;
-    std::map<std::string, ledger::TwitchEventInfo>::const_iterator iter =
-        twitch_events.find(media_key);
-    if (iter != twitch_events.end()) {
-      old_event = iter->second;
+  if (media_id.find("_vod_") != std::string::npos) {
+    // VOD
+    std::vector<std::string> media_props =
+        braveledger_bat_helper::split(media_id, MEDIA_DELIMITER);
+    if (media_props.empty()) {
+      return;
     }
 
-    ledger::TwitchEventInfo new_event(twitch_info);
-    new_event.status_ = GetTwitchStatus(old_event, new_event);
-
-    uint64_t real_duration = GetTwitchDuration(old_event, new_event);
-    twitch_events[media_key] = new_event;
+    std::string media_url = GetMediaURL(user_id);
+    std::string oembed_url =
+        (std::string)TWITCH_VOD_URL + media_props[media_props.size() - 1];
 
-    std::string id = publisher_info->id;
-    auto callback = std::bind(&MediaTwitch::OnSaveMediaVisit,
+    auto callback = std::bind(&MediaTwitch::OnEmbedResponse,
                               this,
+                              real_duration,
+                              media_key,
+                              media_url,
+                              visit_data,
+                              window_id,
+                              user_id,
                               _1,
-                              _2);
+                              _2,
+                              _3);
+
+    const std::string url = (std::string)TWITCH_PROVIDER_URL + "?json&url=" +
+        ledger_->URIEncode(oembed_url);
 
-    ledger_->SaveMediaVisit(id,
-                            updated_visit_data,
-                            real_duration,
-                            window_id,
-                            callback);
+    FetchDataFromUrl(url, callback);
+    return;
   }
+
+  // Live stream
+  SavePublisherInfo(real_duration,
+                    media_key,
+                    GetMediaURL(media_id),
+                    media_id,
+                    visit_data,
+                    window_id,
+                    "",
```

Comment:
Can we please use `std::string()` or `{}`?

`""` still requires significant processing that can't be optimized away. Insignificant, but not if we start doing this all over the place.

Reply:
done